### ISSUE TITLE
GPU Testing Fails on Specific val_acc values

### ISSUE THREAD

**Issue by srb101** [SerpentAI/SerpentAI#115]
* **Operating System:** Windows
* **Serpent Version:** 0.1.12b1
* **Game:** Bunch of Random Screenshots
* **Backend:** GPU

There are times when the model training seems to be stuck at exactly  the same probabilities. Running serpent train context 15 multiple times makes it work.

(Whenever it fails, it's one of the two val_acc: 0.6566 or 0.2323 over and over.

Try 1:
Using TensorFlow backend.
Found 898 images belonging to 4 classes.
Found 99 images belonging to 4 classes.
2017-11-29 02:03:11.861017: I C:\tf_jenkins\home\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\platform\cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX
2017-11-29 02:03:12.442981: I C:\tf_jenkins\home\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1030] Found device 0 with properties:
name: GeForce GTX 1080 major: 6 minor: 1 memoryClockRate(GHz): 1.8095
pciBusID: 0000:01:00.0
totalMemory: 8.00GiB freeMemory: 6.61GiB
2017-11-29 02:03:12.443090: I C:\tf_jenkins\home\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: GeForce GTX 1080, pci bus id: 0000:01:00.0, compute capability: 6.1)
c:\program files\python36\lib\site-packages\serpent\machine_learning\context_classification\context_classifiers\cnn_inception_v3_context_classifier.py:71: UserWarning: The semantics of the Keras 2 argument `steps_per_epoch` is not the same as the Keras 1 argument `samples_per_epoch`. `steps_per_epoch` is the number of batches to draw from the generator at each epoch. Basically steps_per_epoch = samples_per_epoch/batch_size. Similarly `nb_val_samples`->`validation_steps` and `val_samples`->`steps` arguments have changed. Update your method calls accordingly.
  callbacks=callbacks
c:\program files\python36\lib\site-packages\serpent\machine_learning\context_classification\context_classifiers\cnn_inception_v3_context_classifier.py:71: UserWarning: Update your `fit_generator` call to the Keras 2 API: `fit_generator(<keras.pre..., validation_data=<keras.pre..., class_weight="auto", callbacks=[], steps_per_epoch=28, epochs=15, validation_steps=99)`
  callbacks=callbacks
Epoch 1/15
29/28 [===============================] - 20s 680ms/step - loss: 5.1778 - acc: 0.6543 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 2/15
29/28 [===============================] - 13s 437ms/step - loss: 5.4570 - acc: 0.6614 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 3/15
29/28 [===============================] - 13s 439ms/step - loss: 5.4417 - acc: 0.6624 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 4/15
29/28 [===============================] - 14s 474ms/step - loss: 5.4723 - acc: 0.6605 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 5/15
29/28 [===============================] - 13s 464ms/step - loss: 5.5028 - acc: 0.6586 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 6/15
29/28 [===============================] - 13s 437ms/step - loss: 5.6096 - acc: 0.6520 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 7/15
29/28 [===============================] - 13s 443ms/step - loss: 5.3654 - acc: 0.6671 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 8/15
29/28 [===============================] - 13s 437ms/step - loss: 5.2128 - acc: 0.6766 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 9/15
29/28 [===============================] - 14s 467ms/step - loss: 5.5028 - acc: 0.6586 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 10/15
29/28 [===============================] - 14s 467ms/step - loss: 5.2128 - acc: 0.6766 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 11/15
29/28 [===============================] - 13s 436ms/step - loss: 5.2586 - acc: 0.6737 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 12/15
29/28 [===============================] - 13s 463ms/step - loss: 5.8081 - acc: 0.6397 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 13/15
29/28 [===============================] - 14s 468ms/step - loss: 5.7623 - acc: 0.6425 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 14/15
29/28 [===============================] - 14s 468ms/step - loss: 5.2281 - acc: 0.6756 - val_loss: 5.5355 - val_acc: 0.6566
Epoch 15/15
29/28 [===============================] - 13s 443ms/step - loss: 5.4723 - acc: 0.6605 - val_loss: 5.5355 - val_acc: 0.6566
Success! Model was saved to 'datasets/context_classifier.model'

Try 2:
D:\All_Work\serpent-gpu.ai>serpent train context 15
Using TensorFlow backend.
Found 898 images belonging to 4 classes.
Found 99 images belonging to 4 classes.
2017-11-29 02:09:13.206248: I C:\tf_jenkins\home\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\platform\cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX
2017-11-29 02:09:13.745495: I C:\tf_jenkins\home\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1030] Found device 0 with properties:
name: GeForce GTX 1080 major: 6 minor: 1 memoryClockRate(GHz): 1.8095
pciBusID: 0000:01:00.0
totalMemory: 8.00GiB freeMemory: 6.61GiB
2017-11-29 02:09:13.745625: I C:\tf_jenkins\home\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: GeForce GTX 1080, pci bus id: 0000:01:00.0, compute capability: 6.1)
c:\program files\python36\lib\site-packages\serpent\machine_learning\context_classification\context_classifiers\cnn_inception_v3_context_classifier.py:71: UserWarning: The semantics of the Keras 2 argument `steps_per_epoch` is not the same as the Keras 1 argument `samples_per_epoch`. `steps_per_epoch` is the number of batches to draw from the generator at each epoch. Basically steps_per_epoch = samples_per_epoch/batch_size. Similarly `nb_val_samples`->`validation_steps` and `val_samples`->`steps` arguments have changed. Update your method calls accordingly.
  callbacks=callbacks
c:\program files\python36\lib\site-packages\serpent\machine_learning\context_classification\context_classifiers\cnn_inception_v3_context_classifier.py:71: UserWarning: Update your `fit_generator` call to the Keras 2 API: `fit_generator(<keras.pre..., validation_data=<keras.pre..., class_weight="auto", callbacks=[], steps_per_epoch=28, epochs=15, validation_steps=99)`
  callbacks=callbacks
Epoch 1/15
29/28 [===============================] - 17s 595ms/step - loss: 8.7152 - acc: 0.4261 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 2/15
29/28 [===============================] - 13s 436ms/step - loss: 13.0103 - acc: 0.1928 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 3/15
29/28 [===============================] - 14s 466ms/step - loss: 13.0560 - acc: 0.1900 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 4/15
29/28 [===============================] - 14s 468ms/step - loss: 12.7508 - acc: 0.2089 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 5/15
29/28 [===============================] - 14s 466ms/step - loss: 12.7813 - acc: 0.2070 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 6/15
29/28 [===============================] - 14s 466ms/step - loss: 12.6897 - acc: 0.2127 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 7/15
29/28 [===============================] - 13s 464ms/step - loss: 12.5829 - acc: 0.2193 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 8/15
29/28 [===============================] - 14s 466ms/step - loss: 13.0408 - acc: 0.1909 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 9/15
29/28 [===============================] - 13s 465ms/step - loss: 12.8271 - acc: 0.2042 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 10/15
29/28 [===============================] - 13s 436ms/step - loss: 12.7813 - acc: 0.2070 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 11/15
29/28 [===============================] - 13s 464ms/step - loss: 12.8424 - acc: 0.2032 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 12/15
29/28 [===============================] - 14s 467ms/step - loss: 13.0866 - acc: 0.1881 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 13/15
29/28 [===============================] - 13s 435ms/step - loss: 12.9492 - acc: 0.1966 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 14/15
29/28 [===============================] - 13s 436ms/step - loss: 13.1324 - acc: 0.1852 - val_loss: 12.3735 - val_acc: 0.2323
Epoch 15/15
29/28 [===============================] - 13s 464ms/step - loss: 13.0255 - acc: 0.1919 - val_loss: 12.3735 - val_acc: 0.2323
Success! Model was saved to 'datasets/context_classifier.model'

Try 3:
D:\All_Work\serpent-gpu.ai>serpent train context 15
Using TensorFlow backend.
Found 898 images belonging to 4 classes.
Found 99 images belonging to 4 classes.
2017-11-29 02:14:10.209516: I C:\tf_jenkins\home\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\platform\cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX
2017-11-29 02:14:10.730676: I C:\tf_jenkins\home\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1030] Found device 0 with properties:
name: GeForce GTX 1080 major: 6 minor: 1 memoryClockRate(GHz): 1.8095
pciBusID: 0000:01:00.0
totalMemory: 8.00GiB freeMemory: 6.61GiB
2017-11-29 02:14:10.730817: I C:\tf_jenkins\home\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: GeForce GTX 1080, pci bus id: 0000:01:00.0, compute capability: 6.1)
c:\program files\python36\lib\site-packages\serpent\machine_learning\context_classification\context_classifiers\cnn_inception_v3_context_classifier.py:71: UserWarning: The semantics of the Keras 2 argument `steps_per_epoch` is not the same as the Keras 1 argument `samples_per_epoch`. `steps_per_epoch` is the number of batches to draw from the generator at each epoch. Basically steps_per_epoch = samples_per_epoch/batch_size. Similarly `nb_val_samples`->`validation_steps` and `val_samples`->`steps` arguments have changed. Update your method calls accordingly.
  callbacks=callbacks
c:\program files\python36\lib\site-packages\serpent\machine_learning\context_classification\context_classifiers\cnn_inception_v3_context_classifier.py:71: UserWarning: Update your `fit_generator` call to the Keras 2 API: `fit_generator(<keras.pre..., validation_data=<keras.pre..., class_weight="auto", callbacks=[], steps_per_epoch=28, epochs=15, validation_steps=99)`
  callbacks=callbacks
Epoch 1/15
29/28 [===============================] - 17s 602ms/step - loss: 4.0153 - acc: 0.6581 - val_loss: 1.1167 - val_acc: 0.7071
Epoch 2/15
29/28 [===============================] - 13s 438ms/step - loss: 0.2121 - acc: 0.9523 - val_loss: 0.4444 - val_acc: 0.8889
Epoch 3/15
29/28 [===============================] - 14s 467ms/step - loss: 0.3663 - acc: 0.9168 - val_loss: 0.1373 - val_acc: 0.9596
Epoch 4/15
29/28 [===============================] - 13s 436ms/step - loss: 0.6009 - acc: 0.9200 - val_loss: 0.0465 - val_acc: 0.9697
Epoch 5/15
29/28 [===============================] - 13s 436ms/step - loss: 0.0630 - acc: 0.9868 - val_loss: 0.0196 - val_acc: 0.9899
Epoch 6/15
29/28 [===============================] - 13s 465ms/step - loss: 0.0463 - acc: 0.9899 - val_loss: 8.8066 - val_acc: 0.0909
Epoch 7/15
29/28 [===============================] - 13s 463ms/step - loss: 0.3477 - acc: 0.9625 - val_loss: 0.0024 - val_acc: 1.0000
Epoch 8/15
29/28 [===============================] - 14s 468ms/step - loss: 0.0762 - acc: 0.9787 - val_loss: 6.9975e-04 - val_acc: 1.0000
Epoch 9/15
29/28 [===============================] - 14s 467ms/step - loss: 0.1120 - acc: 0.9696 - val_loss: 7.5580e-04 - val_acc: 1.0000
Epoch 10/15
29/28 [===============================] - 13s 443ms/step - loss: 0.0165 - acc: 0.9939 - val_loss: 0.0952 - val_acc: 0.9798
Epoch 11/15
29/28 [===============================] - 13s 464ms/step - loss: 0.0552 - acc: 0.9888 - val_loss: 0.0037 - val_acc: 1.0000
Epoch 12/15
29/28 [===============================] - 14s 469ms/step - loss: 0.2398 - acc: 0.9564 - val_loss: 3.2668e-04 - val_acc: 1.0000
Epoch 13/15
29/28 [===============================] - 13s 437ms/step - loss: 0.0011 - acc: 1.0000 - val_loss: 2.4800e-04 - val_acc: 1.0000
Epoch 14/15
29/28 [===============================] - 13s 436ms/step - loss: 0.5559 - acc: 0.9301 - val_loss: 0.0052 - val_acc: 1.0000
Epoch 15/15
29/28 [===============================] - 13s 436ms/step - loss: 0.0080 - acc: 0.9980 - val_loss: 4.3006e-04 - val_acc: 1.0000
Success! Model was saved to 'datasets/context_classifier.model'

**Comment by nbrochu** [comment]
Thank you for the report but I have no way of reproducing this. I'll keep an eye out for similar scenarios when working with `serpent train context`
